beam-migrate/Database/Beam/Migrate/SQL/BeamExtensions.hs

```diff
@@ -29,6 +29,6 @@ import Data.Text (Text)
 --   'genericSerial' may appear and may fail at run-time if these conditions
 --   aren't met. Please refer to the backend of your choice for more
 --   information.
-class BeamMigrateSqlBackend be =>
-  BeamSqlBackendHasSerial be where
-  genericSerial :: FieldReturnType 'True 'False be (SqlSerial Int) a => Text -> a
+class (BeamMigrateSqlBackend be, Integral n) =>
+  BeamSqlBackendHasSerial n be where
```

Comment:
I'd be in favor of removing the `Integral` constraint here. `Integral` implies `Num` which implies the ability to do arithmetic. However, I can imagine wanting to add support for an opaque ID type which prohibits arithmetic. Backends already have the ability to constrain the valid `n`s in their instances.